champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/brute/passives/Stampede.java

```diff
@@ -116,25 +133,45 @@ public void updateSpeed() {
 
     @EventHandler(priority = EventPriority.HIGHEST)
     public void onDamage(CustomDamageEvent event) {
-        if (event.getCause() != DamageCause.ENTITY_ATTACK) return;
-        if (!(event.getDamager() instanceof Player damager)) return;
-        int str = sprintStr.getOrDefault(damager, -1);
-        if (str < 0) return;
-
-        sprintTime.remove(damager);
-        sprintStr.remove(damager);
-        damager.removePotionEffect(PotionEffectType.SPEED);
-
-        event.setKnockback(false);
-        VelocityData velocityData = new VelocityData(UtilVelocity.getTrajectory2d(damager, event.getDamagee()), 2.0D, true, 0.0D, 0.4D, 1.0D, true);
-        UtilVelocity.velocity(event.getDamagee(), event.getDamager(), velocityData, VelocityType.KNOCKBACK);
-        event.setDamage(event.getDamage() + ((str + 1) * damage));
+
+        if (event.getDamager() instanceof Player damager) {
+
+            if (event.getCause() != DamageCause.ENTITY_ATTACK) return;
+            int str = sprintStr.getOrDefault(damager, 0);
+            if (str < 1) return;
+
+            int level = getLevel(damager);
+
+            sprintTime.remove(damager);
+            sprintStr.remove(damager);
+            damager.removePotionEffect(PotionEffectType.SPEED);
+
+            double bonusKnockback = getBonusKnockback(level);
+
+            event.setKnockback(false);
+            VelocityData velocityData = new VelocityData(UtilVelocity.getTrajectory2d(damager, event.getDamagee()), bonusKnockback, true, 0.0D, 0.4D, 1.0D, true);
+            UtilVelocity.velocity(event.getDamagee(), damager, velocityData, VelocityType.KNOCKBACK);
+
+            double additionalDamage = (str + 1) * getDamage(level);
+            event.setDamage(event.getDamage() + additionalDamage);
+        } else if (event.getDamagee() instanceof Player player) {
+
+            sprintTime.remove(player);
+            sprintStr.remove(player);
+            if(player.hasPotionEffect(PotionEffectType.SPEED)) {
+                player.removePotionEffect(PotionEffectType.SPEED);
+            }
```

Comment:
lol, this will remove speed from anybody who takes damage, not just people with stampede

Reply:
whoopsie dasies